### PR TITLE
Use a common way to generate CollectionName and PartitionKey

### PR DESCRIPTION
**Pull request by alexfca**
This PR paves the way to use different PartitionKey schemes for different collections/tables.
Previously, all tables used a generic cosmosdbapi.GetPartitionKey() to create a PartitionKey which was scoped to the entire Collection. 
As part of the re-partitioning we want to be able to make the PK more specific in some cases for performance reasons.
This PR is a refactor only and has no functional changes but provides a framework to make these changes more transparently.

The main changes in this PR are;
- Make all the Cosmos objects private to the module by renaming to camelCase
- Create 2 methods (getCollectionName() and getPartitionKey()) on each Table to standardize the code
- Remove the specific queryXXXObject() methods on all Tables and replace with a shared PerformQuery() method
- General refactor and bug fixes

### CODE REVIEW

File: appservice/storage/cosmosdb/appservice_events_table.go [hunk 1]
```diff
@@ -45,20 +45,20 @@ import (
 // CREATE INDEX IF NOT EXISTS appservice_events_as_id ON appservice_events(as_id);
 // `
 
-type EventCosmos struct {
+type eventCosmos struct {
```
Comment (alexfca) [inline appservice/storage/cosmosdb/appservice_events_table.go:48]: This makes the cosmos classes private to the module

File: appservice/storage/cosmosdb/appservice_events_table.go [hunk 2]
```diff
@@ -119,50 +119,16 @@ func (s *eventsStatements) prepare(db *Database, writer sqlutil.Writer) (err err
...
-		s.db.cosmosConfig.ContainerName,
-		query,
-		&response,
-		optionsQry)
-
-	if err != nil {
-		return nil, err
-	}
-	return response, nil
+func (s *eventsStatements) getCollectionName() string {
```
Comment (alexfca) [inline appservice/storage/cosmosdb/appservice_events_table.go:122]: These are the new methods added to all tables

File: appservice/storage/cosmosdb/appservice_events_table.go [hunk 3]
```diff
@@ -119,50 +119,16 @@ func (s *eventsStatements) prepare(db *Database, writer sqlutil.Writer) (err err
...
+	return cosmosdbapi.GetCollectionName(s.db.databaseName, s.tableName)
 }
 
-func queryEventEventNumber(s *eventsStatements, ctx context.Context, qry string, params map[string]interface{}) ([]EventNumberCosmosData, error) {
```
Comment (alexfca) [inline appservice/storage/cosmosdb/appservice_events_table.go:143]: This is the removal of the specific query() methods

File: appservice/storage/cosmosdb/appservice_events_table.go [hunk 4]
```diff
@@ -223,21 +189,25 @@ func (s *eventsStatements) selectEventsByApplicationServiceID(
...
 
-	var dbCollectionName = cosmosdbapi.GetCollectionName(s.db.databaseName, s.tableName)
 	params := map[string]interface{}{
-		"@x1": dbCollectionName,
+		"@x1": s.getCollectionName(),
 		"@x2": applicationServiceID,
 	}
 
-	eventRows, err := queryEvent(s, ctx, s.selectEventsByApplicationServiceIDStmt, params)
+	var rows []eventCosmosData
```
Comment (alexfca) [inline appservice/storage/cosmosdb/appservice_events_table.go:197]: This is the new "standard" way to query which calls PerformQuery()

**Review by jaredjn — APPROVED** [review]
Awesome Alex!

**Pull request merged**